### ISSUE TITLE
[MNG-5608] warn if file-based profile activation uses ${project.basedir} since only ${basedir} is supported

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#6705]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MNG-5608](https://issues.apache.org/jira/browse/MNG-5608?redirect=false)** and commented

documentation was improved in MNG-5590

but adding a warning will help users discover the documentation

---

**Affects:** 3.2.1

**Issue Links:**
- [MNG-2363](https://issues.apache.org/jira/browse/MNG-2363) \<profile>\<activation>\<file>\<exists/> does not work in a multi-project build

- [MNG-5590](https://issues.apache.org/jira/browse/MNG-5590) ${basedir} and ${project.basedir} do not behave the same on file-based profile activation

- [MNG-5612](https://issues.apache.org/jira/browse/MNG-5612) effective pom should not contain file-based profile activation fully interpolated but with limited interpolation like seen during profile activation

- [MNG-5516](https://issues.apache.org/jira/browse/MNG-5516) ${project.basedir} in profile activation exists clause strange behavior

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

warning added in [64c41950](http://git-wip-us.apache.org/repos/asf/maven/commit/64c41950)
IT added in [f15c2022](http://git-wip-us.apache.org/repos/asf/maven-integration-testing/commit/f15c2022)

sample warning text from IT execution:

```
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.maven.its.mng5608:profile-test:pom:1
[WARNING] Failed to interpolate file location ${project.basedir}/pom.xml for profile mng-5608-exists-project.basedir: ${project.basedir} expression not supported during profile activation, use ${basedir} instead @ line 56, column 19
[WARNING] Failed to interpolate file location ${project.basedir}/pom.xml for profile mng-5608-missing-project.basedir: ${project.basedir} expression not supported during profile activation, use ${basedir} instead @ line 67, column 20
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
```

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Code should be moved to the DefaultModelValidator

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [06a7d6dd](http://git-wip-us.apache.org/repos/asf/maven/commit/06a7d6dd)
